### ISSUE TITLE
vibe.d won't build - GenericPath's constructor is not nothrow

### ISSUE THREAD

**Issue by the-Mitu** [vibe-d/vibe.d#2184]
../../.dub/packages/vibe-core-1.4.1/vibe-core/source/vibe/core/path.d(389,7): Error: constructor `vibe.core.path.GenericPath!(InetPathFormat).GenericPath.__ctor!(OnlyResult!(Segment, 1LU)).this` is not `nothrow`
    ../../.dub/packages/vibe-core-1.4.1/vibe-core/source/vibe/core/path.d(386,2): Error: `nothrow` constructor `vibe.core.path.GenericPath!(InetPathFormat).GenericPath.this` may throw
    ../../.dub/packages/vibe-core-1.4.1/vibe-core/source/vibe/core/path.d(194,18): Error: template instance `vibe.core.path.GenericPath!(InetPathFormat)` error instantiating
    /usr/bin/dmd failed with exit code 1.

Compilation fails. The same error on both D 2.080.1 and 2.081.0 on Ubuntu 18.04.

**Comment by wilzbach** [comment]
See https://github.com/vibe-d/vibe.d/issues/2181 for build failures with 2.081 (note there's already a patch beta out and the patch release will be out soon.)

Now I haven't seen your issue nor can I reproduce this with 2.080 or 2.081 in the vibe.d repository (and Vibe.d does pass its testsuite for 2.081.1-beta.1 again).

What project are you trying to build (and how)?

**Comment by the-Mitu** [comment]
Not the one from example, but my own.

src/simple.d
```
module simple.service;

import vibe.vibe;

void index(HTTPServerRequest, HTTPServerResponse res)
{
	res.writeBody("Wow, that works!");
}

void main()
{
	// Testing code
	auto router = new URLRouter;
	router.get("/", &index);

	auto settings = new HTTPServerSettings;
	settings.port = 8080;

	listenHTTP(settings, router);

	// Testing code end

	lowerPrivileges();
	runEventLoop();
}
```

dub.json
```
{
	"name": "simple",
	"authors": [
		"Tester"
	],
	"dependencies": {
		"vibe-d": "~>0.8.4"
	},
	"description": "Test",
}
```

**Comment by the-Mitu** [comment]
I've just tried to build the app_skeleton sample with D 2.080.1 and got the same error.

**Comment by wilzbach** [comment]
So I think this due to vibe-core 1.4.1 being released two hours ago (https://github.com/vibe-d/vibe-core/commit/20558e44167d730df3e169d67e0cdcf728af7a0f) and this release brings the regression.
As a workaround you can switch back to libevent for now (or explicitly downgrade vibe-core in your dub.selections.json).

```d
#!/usr/bin/env dub
/+dub.sdl:
dependency "vibe-d" version="~>0.8.4"
subConfiguration "vibe:core" "libevent"
+/
import vibe.vibe;

void index(HTTPServerRequest, HTTPServerResponse res)
{
	res.writeBody("Wow, that works!");
}

void main()
{
	// Testing code
	auto router = new URLRouter;
	router.get("/", &index);

	auto settings = new HTTPServerSettings;
	settings.port = 8080;

	listenHTTP(settings, router);

	// Testing code end

	lowerPrivileges();
	runEventLoop();
}
```

CC @s-ludwig

**Comment by Kozzi11** [comment]
@the-Mitu no you do nothing wrong, I have same issue, problem is with this commit:
https://github.com/vibe-d/vibe-core/commit/5f18693c49a0c0a7359bb32fefd5f8719950a573

**Comment by wilzbach** [comment]
FYI: https://github.com/vibe-d/vibe-core/pull/85

**Comment by the-Mitu** [comment]
Downgrading vibe-core helped Thank you :)

**Comment by s-ludwig** [comment]
Seems I rushed this a bit. Corrected now by fae7d3e93d00d9636632aa0acf9ebc19ed9f4a34. Sorry @wilzbach, I didn't keep up with the GH notifications and missed your early fix.
